addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/NPMContentAccessHandler.java

```diff
@@ -0,0 +1,498 @@
+            }
+
+            final URI uri = uriBuilder.get();
+
+            Response.ResponseBuilder builder = Response.created( uri );
+            if ( builderModifier != null )
+            {
+                builderModifier.accept( builder );
+            }
+            response = builder.build();
+
+            // generate .http-metadata.json for hosted repo to resolve npm header requirements
+            generateHttpMetadataHeaders( tomerge, request, response );
+        }
+        catch ( final IndyWorkflowException | IOException e )
+        {
+            logger.error( String.format( "Failed to upload: %s to: %s. Reason: %s", path, name, e.getMessage() ), e );
+
+            response = formatResponse( e, builderModifier );
+        }
+        finally
+        {
+            closeQuietly( stream );
+        }
+
+        return response;
+    }
+
+    @Override
+    public Response doGet( String packageType, String type, String name, String path, String baseUri,
+                           HttpServletRequest request, EventMetadata eventMetadata )
+    {
+        return doGet( packageType, type, name, path, baseUri, request, eventMetadata, null );
+    }
+
+    @Override
+    public Response doGet( String packageType, String type, String name, String path, String baseUri,
+                           HttpServletRequest request, EventMetadata eventMetadata,
+                           Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        if ( !PackageTypes.contains( packageType ) )
+        {
+            Response.ResponseBuilder builder = Response.status( 400 );
+            if ( builderModifier != null )
+            {
+                builderModifier.accept( builder );
+            }
+            return builder.build();
+        }
+
+         // hide npm sensitive user info for publish
+         if ( path.startsWith( "-/user" ) )
+         {
+             Response.ResponseBuilder builder = Response.status( 404 );
+             if ( builderModifier != null )
+             {
+                builderModifier.accept( builder );
+             }
+             return builder.build();
+         }
+
+        final StoreType st = StoreType.get( type );
+        final StoreKey sk = new StoreKey( packageType, st, name );
+
+        eventMetadata = eventMetadata.set( ContentManager.ENTRY_POINT_STORE, sk );
+
+        final AcceptInfo acceptInfo = jaxRsRequestHelper.findAccept( request, ApplicationContent.text_html );
+        final String standardAccept = ApplicationContent.getStandardAccept( acceptInfo.getBaseAccept() );
+
+        Response response = null;
+
+        logger.info(
+                "GET path: '{}' (RAW: '{}')\nIn store: '{}'\nUser addMetadata header is: '{}'\nStandard addMetadata header for that is: '{}'",
+                path, request.getPathInfo(), sk, acceptInfo.getRawAccept(), standardAccept );
+
+        if ( path == null || path.equals( "" ) || request.getPathInfo().endsWith( "/" ) || path.endsWith(
```

Comment:
as this "path" is used upper with startsWith, so here it will always be not null. So maybe a null check is also needed before the startsWith.

Reply:
@gli, just same as the ContentAccessHandler, simply do that to prevent some NPE may extend in future?

Reply:
Before this compare, you called path.startsWith once(line 187?), so if path can be null, it has caused NPE, and will not trigger this compare.